lib/Ast.ml

```diff
@@ -1985,6 +1985,7 @@ end = struct
      |Cl {pcl_desc= Pcl_fun _; _}, Ppat_tuple _
      |Cl {pcl_desc= Pcl_fun _; _}, Ppat_construct _
      |Cl {pcl_desc= Pcl_fun _; _}, Ppat_alias _
+     |Cl {pcl_desc= Pcl_fun _; _}, Ppat_lazy _
```

Comment:
Some more cases are missing:
- `Ppat_exception`
- `Ppat_variant` with arguments
- `Ppat_or`

I wonder if a refactor of this function is needed. What do you thing of classifying the context argument separately (eg. to `` Patern of pattern | Function | Let | Parameter ``), so we have saner rules that can't forget any cases ?

Reply:
I don't think that Ppat_exception can be found here, but yes this looks more complicated than it should. To be fair, I'm not sure I understand why `parenze_x` functions are structured this way, compared to assigning precedence levels and comparing them.

Reply:
I think these functions are this way because we did let that happen :( Now it's time to improve them !
Computing precedence would be great but a lot more work.

Reply:
OK I'll have a look at this.

Reply:
it's likely that we can improve, and @CraigFe has started in this direction, but this is not strictly related to this PR